php/libraries/Data/DTO.php

```diff
@@ -0,0 +1,22 @@
+<?php
+namespace LORIS\Data;
+
+abstract class DTO
+{
+    public function __toString()
+    {
+        return json_encode($this->asArray());
+    }
+
+    public function asArray()
+    {
+        // get all object vars
+        $object_vars = get_object_vars($this);
```

Comment:
This line will mean that you can't have private/helper attributes that don't get serialized.

Maybe there should be something like an `abstract protected function ModelFields() : array` which tells you which columns should be serialized and filters the object_vars down to those?

(Or maybe I'm misunderstanding the point of this class.)

Reply:
It can be changed. It was only a short-cut for prototyping